src/mozanalysis/metrics.py

```diff
@@ -320,6 +376,30 @@ def get_sanity_metrics(self, experiment_slug: str) -> list[Metric]:
         else:
             raise ValueError
 
+    @classmethod
+    def from_mcp_data_source(
+        cls,
+        parser_data_source: ParserDataSource,
+        app_name: str | None = None,
+        group_id_column: str | None = ExperimentalUnit.PROFILE_GROUP.value,
+    ) -> DataSource:
```

Comment:
I was going to say that I don't love the `mcp_data_source` naming here, but realized I don't have a good solution. So instead I'll just note that I wish we didn't have the same (or nearly the same) objects defined all over the place.  I guess another symptom of the many-repo issue we've discussed at length..

Reply:
Yeah, I agree with you. Some ideas I've got: 
- rename mcp objects into "definition"s, then mozanalysis objects can have a "from_definition" method
- instead of mozanalysis's DataSource largely duplicating the parameters, maybe it should wrap the mcp object and only take the extra parameters it needs which, afaict, is just app_id (which isn't present in the data source definition and is provided by jetstream at runtime)

Reply:
I think I like the second one, and maybe we call the mozanalysis version `ExperimentDataSource` or something? What do you think?

Either way I'm OK with leaving as-is in this PR, better to separate out that complexity.

Reply:
+1. I also like the 2nd one better, and agree with doing it later, not in this PR

Reply:
Issue filed: https://github.com/mozilla/mozanalysis/issues/293